### PR TITLE
Add Artifacts and Split run.sh jobs

### PR DESCRIPTION
**Pull request by fridiculous**
This PR addresses issue https://github.com/covid-projections/covid-data-model/issues/< >

### Please Check
- [ ] Will the *current* schema in [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.md) be updated with this PR?
- [ ] Are tests passing?

### CODE REVIEW

File: .github/workflows/deploy_api.yml [hunk 1]
```diff
@@ -9,24 +9,131 @@ on:
   schedule:
    - cron: '30 0,12 * * *'
 
+  push:
```
Comment (fridiculous) [inline .github/workflows/deploy_api.yml:12]: wil remove this, once the entire pipeline is ready

File: .github/workflows/deploy_api.yml [hunk 2]
```diff
@@ -9,24 +9,131 @@ on:
...
 
 jobs:
-  publish-api:
+  execute-model:
     runs-on: self-hosted
+    env:
+      # To pin to an old data set or model code, put the branch/tag/commit here:
+      COVID_DATA_PUBLIC_REF: 'master'
+      # !!! Change this to your BRANCH if you want to test it
+      COVID_DATA_MODEL_REF: 'fridiculous/add-artifacts'
```
Comment (fridiculous) [inline .github/workflows/deploy_api.yml:25]: will revert this to master once the entire pipeline is ready

File: .github/workflows/deploy_api.yml [hunk 3]
```diff
@@ -9,24 +9,131 @@ on:
...
+      run: pip install -r requirements.txt
 
+    # - name: create temp file
+    #   run: |
+    #     mkdir -p /data/api-results-${{env.SNAPSHOT_ID}}
+    #     touch /data/api-results-${{env.SNAPSHOT_ID}}/deleteme
+    - name: Build Model Results (run.sh .. .. execute_model)
+      env:
+        COVID_MODEL_CORES: 96
+      run: ./covid-data-model/run.sh ./covid-data-public /data/api-results-${{env.SNAPSHOT_ID}} execute_model
```
Comment (ghop02) [inline .github/workflows/deploy_api.yml:65]: I think this code might be cleaner with some more environment variables!
```

env: 
  COVID_DATA_PUBLIC: ./covid-data-public
run: ./covid-data-model/run.sh ${COVID_DATA_PUBLIC}
```

Also, you can add top-level environment variables in the workflow if that's helpful!

File: .github/workflows/deploy_api.yml [hunk 4]
```diff
@@ -41,7 +148,60 @@ jobs:
...
+
+    # - uses: actions/download-artifact@v1
+    #   with:
+    #     name: model-results
+    #     path: ./api-results
+    - name: Build DoD (run.sh .. .. execute_summaries)
+      env:
+        COVID_MODEL_CORES: 96
+      run: ./covid-data-model/run.sh ./covid-data-public /data/api-results-${{env.SNAPSHOT_ID}} execute_dod
+    # - name: Upload DoD with Model Results
```
Comment (ghop02) [inline .github/workflows/deploy_api.yml:176]: Why are these commented out?

File: .gitignore [hunk 5]
```diff
@@ -1,6 +1,8 @@
 .~lock.*
 results/*
 api_results/*
+api-results*/*
```
Comment (ghop02) [inline .gitignore:4]: why do we have api-results and api_results?

File: .github/workflows/deploy_api.yml [hunk 6]
```diff
@@ -9,24 +9,131 @@ on:
...
+      working-directory: ./covid-data-model
+      run: pip install -r requirements.txt
+
+    # - name: create temp file
+    #   run: |
+    #     mkdir -p /data/api-results-${{env.SNAPSHOT_ID}}
+    #     touch /data/api-results-${{env.SNAPSHOT_ID}}/deleteme
+    - name: Build Model Results (run.sh .. .. execute_model)
+      env:
+        COVID_MODEL_CORES: 96
```
[the remark this answers is not in the document]
Comment (fridiculous) [reply .github/workflows/deploy_api.yml:64]: It is determined in python or set by an env var.  it's intentionally overridden, to differentiate CI servers

File: .gitignore (hunk 5 above)
Comment (fridiculous) [reply .gitignore:4]: new runs have `api-results-#` as a format.  keeping others for backwards compatability

File: .github/workflows/deploy_api.yml (hunk 4 above)
Comment (fridiculous) [reply .github/workflows/deploy_api.yml:176]: for future debugging, but maybe worth cleaning up later

File: .github/workflows/deploy_api.yml (hunk 3 above)
Comment (fridiculous) [reply .github/workflows/deploy_api.yml:65]: I'm not sure i follow.

File: run.sh [hunk 7]
```diff
@@ -9,16 +9,26 @@ set -o errexit
 # Checks command-line arguments, sets variables, etc.
 prepare () {
   # Parse args if specified.
-  if [ $# -ne 2 ]; then
-    echo "Usage: $0 [covid-data-public directory] [output-directory]"
+  if [ $# -lt 2 ] || [ $# -gt 3 ]; then
+    echo "Usage: $0 [covid-data-public directory] [output-directory] (optional - specific function)"
```
[the remark this answers is not in the document]
Comment (fridiculous) [reply run.sh:13]: yea, i agree it's not clear.  although i don't want to insert too much change on the  default runs atm

**Pull request merged**